provisioning/roles/nginx/templates/letsencrypt.j2

```diff
@@ -0,0 +1,15 @@
+# Initial configuration for Let's Encrypt.
+# You should not use this configuration as the definitive one.
```

Comment:
I think it's fine to have *some* dependency between the letsencrypt and nginx roles, since letsencrypt needs a working http server to generate certificates anyways. Note that currently the nginx role somewhat depends on the letsencrypt role since it contains stuff about the `well-known` directory. What I would do is have the nginx role handle letsencrypt certificate generation and:

 - Place the HTTP and HTTPS config in the `sites-available` directory. If the variable `use_letsencrypt` is set, the HTTP config should contain the `.well-known` setup and the HTTPS config should contain paths to the letsencrypt certificates; otherwise the HTTPS config will use the snakeoil certificates.
 - Enable the HTTP config
 - If the `use_letsencrypt` variable is set, setup letsencrypt renewal and generate a certificate.
 - Enable the HTTPS config

What do you think?